documentation/add/01_pace_framework.md

```diff
@@ -0,0 +1,145 @@
+together with interface code which allows a very similar syntax to Matlab for Python users.
+In addition, `pace-python` currently also packages the Python part of `brillem`
+(it cannot use the Matlab part of `brillem`).
+
+
+## Call-graph
+
+![PACE project call-graph](diagrams/pace_framework.svg)
+
+The above diagram shows the call graphs between the PACE and interface programs.
+Arrows point from the caller to the callee (e.g. Horace calls `euphonic_matlab`),
+and the text beside the lines indicate the called function or method.
+Purple background indicates major PACE packages.
+
+The code is organised such that each (yellow-background) file in the graph is a separate git repository,
+except for the "compiled" versions included in `pace-python` which are all in a single
+[repository](https://github.com/pace-neutrons/pace-python/).
+
+
+## `euphonic_horace`
+
+The `euphonic_horace` module is thought to be a little superfluous and it has been proposed to remove it.
+This could be done in a few different ways:
+
+* Remove `euphonic_horace`, splitting and duplicating the code in  `pace-python` and Horace.
+    - Pro: one less package which can confuse users, and perhaps simplify the integration pipeline.
+    - Con: code duplication
+    - Con: Horace is a large program with a long test time, changes to the Euphonic interface would require more developer effort 
+      than in a separate repository.
+* Remove `euphonic_horace`, putting the code into `euphonic_matlab`.
+    - Pro: one less package which can confuse users, and perhaps simplify the integration pipeline.
+    - Con: `euphonic_matlab` would have both Python and Matlab code and needs two different versions for each language.
+        + git allows only a single version tag so would need to enfore a mechanism where Matlab code changes
+          would change the subminor version number whilst Python code changes increment the minor or major version number.
+    - Con: `pace-python` would download the `euphonic_matlab` package and extract the Python code (it does not need the Matlab code).
+        + makes the build script slightly more complex (but not much).
+* Remove `euphonic_horace`, moving the current Python code into `pace-python` and resurrect/reimplement a Matlab equivalent in `euphonic_matlab`.
+    - Pro: one less package which can confuse users, and perhaps simplify the integration pipeline.
+    - Con: code duplication (have different Python and Matlab implementations of the same functionality)
+    
+
+## Naming
+
+Some project names, particularly `pace-python`, `euphonic_horace` and `euphonic_matlab` are thought to be lacking.
+Some suggested alternatives are:
+
+* For `pace-python`: `snakeskin`, `pacy`, `pace-neutron`
+* For `euphonic_horace`: `euphonic_horace_interface_python`, `euphonic_sqw`, `euphonic_horace_driver`
+* For `euphonic_matlab`: `euphonic_horace_interface_matlab`, `euphonium`, `meuphonic`
+
```

Comment:
The discussion part of this document is dangling a bit. Ideally we would recrod decisions on euphonic-horace and on naming. For the moment I think it is acceptable to decide to leave things as-is now and revisit later e.g. after PACE, but we should record that decision here.

Reply:
Yeah... it petered out a bit... I've added a comment on the naming below this section.